File: services/shop-backend/src/main/java/io/miragon/shop/application/port/outbound/ArticleRepository.java
package io.miragon.shop.application.port.outbound;

import io.miragon.shop.domain.article.Article;
import io.miragon.shop.domain.article.ArticleId;

import java.util.List;

public interface ArticleRepository {
    List<Article> loadAll();
    List<Article> loadByIds(List<ArticleId> articleIds);
    void save(Article article);
    Article findById(ArticleId articleId);
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/application/port/outbound/ArticleRepository.kt
package io.miragon.shop.application.port.outbound

import io.miragon.shop.domain.article.Article
import io.miragon.shop.domain.article.ArticleId

interface ArticleRepository {
    fun loadAll(): List<Article>
    fun loadByIds(articleIds: List<ArticleId>): List<Article>
    fun save(article: Article)
    fun findById(articleId: ArticleId): Article?
}